src/com/sun/jna/NativeLibrary.java

```diff
@@ -947,6 +947,9 @@ else if (Platform.isARM()) {
             cpu = "arm";
             libc = "-gnueabi";
         }
+        else if (Platform.isMIPS()) {
+            cpu = (Platform.is64Bit() ? "mips64el" : "mipsel");
+        }
```

Comment:
`cpu` is filled from `Platform#getCanonicalArchitecture`, so is this needed or does the JVM set the correct architecture? I would remove this, as this might also conflict with big-endian mips support.

I checked debian mips64el and there is also a different libc suffix: `-gnueabi64`:
`/usr/lib/mips64el-linux-gnuabi64`
So libc should be corrected. I suggest:

```java
        else if (Platform.ARCH.equals("mips64el")) {
            libc = "-gnueabi64";
        }
```